docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -463,6 +463,33 @@ router's MAC address (e.g., `fe80::858:cbff:fecb:1`). This LLA
 remains stable as a default gateway, independent of the virtual
 machine's hosting node.
 
+#### Tunnel keys for the transit switches/routers
+
+OVN has a reserved range of 2^16 for the transit datapaths https://github.com/ovn-org/ovn/commit/969f7f54ea3868172f99119991d8d875bb8f240c
+that won't be assigned by the northd for the regular datapaths.
+We already use this key for the Layer3 transit switches. So currently we can support not more than 2^16 (65K) layer3 UDNs.
+Now we need more transit routers for the layer2 topology, but the allowed tunnel key range stays the same.
+The same problem will arise when we get to UDN interconnect, because it will use even more transit routers.
+
+For this enhancement we have 2 options:
+- We could just split the range into 2 parts, and say we support 32K layer3 UDNs and 32K layer2 UDNs. UDN interconnect
+could do later what is described in the option 3.
+- Second option is to use tunnel-keys outside the protected range. Current range is split into [0, 2^24-2^16-1] for northd
+allocation and [2^24-2^16, 2^24-1] for interconnect. Northd does sequential key allocation, that means
+(for single-node-zone) it will only use (`Number of UDNs` + 1)*5 keys, so the following range [`Number of UDNs` + 1)*5; 2^24-2^16-1]
+will be unused, and we could start allocating transit router keys from 2^24-2^16-1 going down. This is kind of dangerous
+as it only relies on the current northd behaviour.
+UDN interconnect could do later what is described in the option 3.
+- Third approach is to start assigning all datapath keys ourselves (for all switches and routers, currently it is done by the northd),
```

Comment:
cant we just defer this problem until later, and then if we ever get to a point where we max out number of UDNs we switch to allocate ourselves, and just preserve the IDs already handed out by northd?

Reply:
yeah that is options 1 and 2. But we still need to decide how to allocate transit router keys (option 1 is the easiest way if we are fine with 32K of layer2 and layer2 UDNs for now)

Reply:
Can you call that out explicitly in the enhancement? It looks like you say that northd automatically allocates for switches, but I don't see anything about routers.